grep/hemartin/grep.go

```diff
@@ -0,0 +1,94 @@
+package grep
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"regexp"
+)
+
+const printLinesFlag = "-n"
+const printNamesFlag = "-l"
+const caseInsensitiveFlag = "-i"
+const invertFlag = "-v"
+const matchLinesFlag = "-x"
+
+func Search(pattern string, flags, files []string) []string {
+	result := []string{}
+
+	printLines := sliceContains(flags, printLinesFlag)
```

Comment:
One comment and that applies to my solution as well, cli tools usually use loop with switch/case to parse the argument in one go. I saw that in Chris's solution and recalled it's a common pattern.